packages/react/src/responsive-image.tsx

```diff
@@ -0,0 +1,219 @@
+      }
+      const sources = widths.map((width) => {
+        const url = props.src.imageUrlFor(width, type);
+        return `${url} ${width}w`;
+      });
+
+      return {
+        srcset: sources.join(', '),
+        sizes: props.sizes ?? (props.size ? `${props.size}vw` : undefined),
+        type,
+        mimeType: `image/${type}`,
+      };
+    });
+  }
+
+  const { width } = props;
+  if (width === undefined) {
+    return [];
+  }
+
+  return props.src.imageTypes.map((type) => {
+    const sources = pixelDensities
+      .map((density) => {
+        const url = props.src.imageUrlFor(width * density, type);
+        return `${url} ${density}x`;
+      })
+      .filter((source) => source !== undefined);
+
+    return {
+      srcset: sources.join(', '),
+      type,
+      mimeType: `image/${type}`,
+    };
+  });
+}
+
+function getWidth(props: ResponsiveImageProps) {
+  const layout = getLayout(props);
+  if (layout === 'responsive') {
+    return getDestinationWidthBySize(props.size);
+  }
+  if (props.width !== undefined) {
+    return props.width;
+  }
+
+  const aspectRatio = props.src.aspectRatio;
+  if (aspectRatio && props.height !== undefined) {
+    return props.height * aspectRatio;
+  }
+
+  return undefined;
+}
+
+function getHeight(props: ResponsiveImageProps) {
+  if (props.height !== undefined) {
+    return props.height;
+  }
+
+  const width = getWidth(props);
+  const aspectRatio = props.src.aspectRatio;
+  if (aspectRatio && width !== undefined) {
+    return Math.round(width / aspectRatio);
+  }
+
+  return undefined;
+}
+
+function getSrc(props: ResponsiveImageProps) {
+  const width = getWidth(props) || 640;
+  return props.src.imageUrlFor(width);
+}
+
+export function ResponsiveImage(props: ResponsiveImageProps) {
+  const layout = getLayout(props);
+  const sources = getSources(props);
+
+  const [isLoaded, setLoaded] = useState(false);
+
+  function getClassNames(props: ResponsiveImageProps) {
+    const classNames = [
+      'ri-img',
+      `ri-${layout === 'responsive' ? 'responsive' : 'fixed'}`,
+    ];
+    const lqipClass = props.src.lqip?.class;
+    if (lqipClass && !isLoaded) {
+      classNames.push(getValueOrCallback(lqipClass));
+    }
+    return classNames.join(' ');
+  }
+
+  function bgImage() {
+    if (isLoaded) {
+      return undefined;
+    }
+    const bgImage = props.src.lqip?.bgImage;
+    return bgImage ? `url("${getValueOrCallback(bgImage)}")` : undefined;
+  }
+
+  return (
+    <picture>
+      {sources
+        .sort(
+          (a, b) => (typeScore.get(b.type) ?? 0) - (typeScore.get(a.type) ?? 0),
+        )
+        .map((s) => (
+          <source srcSet={s.srcset} type={s.mimeType} sizes={s.sizes} />
+        ))}
+      <img
+        className={getClassNames(props)}
+        loading={props.loading || 'lazy'}
+        decoding={props.decoding || 'async'}
+        alt={props.alt}
+        width={getWidth(props)}
+        height={getHeight(props)}
+        {...props}
```

Comment:
Wouldn't this apply all props as HTMl attributes, even our own ones that are not valid attributes? 

If so, I think we need to split our props and apply only the ...rest of it as attributes, as in [this example](https://github.com/simonihmig/responsive-image/blob/main/packages/svelte/src/lib/responsive-image.svelte#L12-L19)